### ISSUE TITLE
FOREACH CASE /MERGE after query/match using apoc.load.jsonParams generates ExecutionFailed error

### ISSUE THREAD

**Issue by pdrangeid** [neo4j/neo4j#12182]
- Neo4j version: 3.5.3
- Operating system: Windows 10
- API/Driver: Neo4J Browser

If I run this with the "FOREACH" line commented out, it runs just fine. If I leave the FOREACH in I receive this error:
Neo.DatabaseError.Statement.ExecutionFailed: Expressions need to be registered with it's owning Pipe, so the profiling knows where to report db-hits

### Steps to reproduce
Run a query as below and attempt to perform FOREACH case evaluation to perform a MERGE
```
MATCH (n:Clientdomain)--(a:Company {name:'Some Company'})--(kba:Knowbe4account)
OPTIONAL MATCH (cnp:Contact {name:'None Provided'})
WITH a,kba,cnp,"super-secret-token" as token,"https://us.api.knowbe4.com/v1/users" as url
CALL apoc.load.jsonParams(url,{Authorization:"Bearer "+token,Accept:"application/json"},null) yield value
OPTIONAL MATCH (c:Contact)--(a) where toLower(trim(value.email)) = toLower(trim(c.email))
OPTIONAL MATCH (c2:Contact)--(a) where toLower(trim(c.name)) = toLower(trim(value.first_name))+' '+toLower(trim(value.last_name))
FOREACH (ignoreMe in CASE WHEN not((kba)--(c)) and exists(c.acctrecid) THEN [1] ELSE [] END | MERGE (kba)-[:KNOWBEFORE_USER]-(c))
return c,kba
```

### Expected behavior
MATCH results will run against the CASE validation and MERGE where properties exisat

### Actual behavior
The following Cypher exception is generated:
Neo.DatabaseError.Statement.ExecutionFailed: Expressions need to be registered with it's owning Pipe, so the profiling knows where to report db-hits

**Comment by Hunterness** [comment]
Hi, thanks for reporting, we will look into it.

/Team Cypher

**Comment by craigtaverner** [comment]
Is it possible to provide a simplified query with sample data so we can reproduce this?

**Comment by sherfert** [comment]
This will be fixed in the next patch releases of 3.4 and 3.5
